tests/testthat/test-seasonal_burden_levels.R

```diff
@@ -0,0 +1,142 @@
+test_that("Test that input argument checks work", {
+
+  start_date <- as.Date("2021-01-04")
+  end_date <- as.Date("2023-12-31")
+
+  weekly_dates <- seq.Date(from = start_date,
+                           to = end_date,
+                           by = "week")
+
+  obs <- stats::rpois(length(weekly_dates), 1000)
+
+  tsd_data <- tsd(
+    observed = obs,
+    time = as.Date(weekly_dates),
+    time_interval = "week"
+  )
+
+  expect_error(
+    checkmate_err_msg(seasonal_burden_levels(tsd_data, decay_factor = 3),
+      "Variable 'decay_factor': Element 1 is not <= 1."
+    )
+  )
+
+  expect_error(
+    checkmate_err_msg(seasonal_burden_levels(tsd_data, n_peak = c(1, 2)),
+      "Variable 'n_peak': Must have length 1, but has length 2."
+    )
+  )
+
+  expect_error(
+    checkmate_err_msg(seasonal_burden_levels(tsd_data, disease_threshold = "hey"),
+      "'disease_threshold': Must be of type 'integerish', not 'character'."
+    )
+  )
+
+  expect_error(
+    checkmate_err_msg(seasonal_burden_levels(tsd_data, season_weeks = c(2, 10)),
+      "`start` must be greater than `end`!"
+    )
+  )
+
+  expect_error(
+    checkmate_err_msg(seasonal_burden_levels(tsd_data, method = "peak_levels"),
+      "Variable 'conf_levels': Must have length 3, but has length 1."
+    )
+  )
+
+  expect_error(
+    checkmate_err_msg(seasonal_burden_levels(tsd_data, conf_levels = c(0.2, 0.5)),
+      "Variable 'conf_levels': Must have length 1, but has length 2."
+    )
+  )
```

Comment:
Just a note: 

If you think that doing these checkmate checks are tedious work (which it definitely can be), you can just skip them I would not require them to accept the PRs.